rskj-core/src/test/java/co/rsk/RskContextTest.java

```diff
@@ -43,25 +60,150 @@ public void getCliArgsSmokeTest() {
     }
 
     @Test
-    public void getBuildInfoSmokeTest() {
-        RskTestContext rskContext = new RskTestContext(new String[0]);
-        mockBuildInfoResource(new ByteArrayInputStream("build.hash=c0ffee\nbuild.branch=HEAD".getBytes()));
-        assertThat(rskContext.getBuildInfo(), notNullValue());
-        assertThat(rskContext.getBuildInfo().getBuildHash(), is("c0ffee"));
+    public void shouldBuildSimpleTrieStore() throws IOException {
+        Path testDatabasesDirectory = databaseDir.getRoot().toPath();
+        doReturn(new GarbageCollectorConfig(false, 1000, 3)).when(testProperties).garbageCollectorConfig();
+        doReturn(testDatabasesDirectory.toString()).when(testProperties).databaseDir();
+
+        RskContext rskContext = new RskContext(new String[0]) {
+            @Override
+            public RskSystemProperties getRskSystemProperties() {
+                return testProperties;
+            }
+        };
+
+        TrieStore trieStore = rskContext.getTrieStore();
+        Assert.assertThat(trieStore, is(instanceOf(TrieStoreImpl.class)));
+        Assert.assertThat(Files.list(testDatabasesDirectory).count(), is(1L));
+    }
+
+    @Test
+    public void shouldBuildSimpleTrieStoreCleaningUpMultiTrieStore() throws IOException, InterruptedException {
+        Path testDatabasesDirectory = databaseDir.getRoot().toPath();
+        doReturn(new GarbageCollectorConfig(false, 1000, 3)).when(testProperties).garbageCollectorConfig();
+        doReturn(testDatabasesDirectory.toString()).when(testProperties).databaseDir();
+
+        RskContext rskContext = new RskContext(new String[0]) {
+            @Override
+            public RskSystemProperties getRskSystemProperties() {
+                return testProperties;
+            }
+        };
+
+        long preExistingEpochs = 4;
+        for (int i = 0; i < preExistingEpochs; i++) {
+            Files.createDirectory(testDatabasesDirectory.resolve(String.format("unitrie_%d", i)));
+        }
+
+        Assert.assertThat(Files.list(testDatabasesDirectory).count(), is(preExistingEpochs));
+        TrieStore trieStore = rskContext.getTrieStore();
+        Assert.assertThat(trieStore, is(instanceOf(TrieStoreImpl.class)));
+        Assert.assertThat(Files.list(testDatabasesDirectory).count(), is(1L));
+    }
+
+    @Test
+    public void shouldBuildMultiTrieStore() throws IOException {
```

Comment:
Please add one test case when multiple folders already exist and the method getTrieStore should build a multi trie store. The other cases are covered :+1: